### PR TITLE
Improve SubmitError show instance

### PR DESCRIPTION
**Pull request by carlpulley-da**
- [x] add in missing data to `SubmitError` show instance

This PR largely resolves https://github.com/digital-asset/daml/issues/17188.

Following work from issue #17188 is not intended to be done on this PR:
> ... and (if possible) the error messages vaguely line up with what the scenario service would show.

### CODE REVIEW

File: daml-script/daml3/Daml/Script/Questions/Submit/Error.daml [hunk 1]
```diff
@@ -121,34 +122,38 @@ data SubmitError
...
     TemplatePreconditionViolated {} -> "Template precondition was violated"
-    CreateEmptyContractKeyMaintainers {} -> "Attempted to create a contract with empty key maintainers"
-    FetchEmptyContractKeyMaintainers {} -> "Attempted to fetch a contract with empty key maintainers"
-    WronglyTypedContract _ expected actual ->
-      "Excepted contract to be a \"" <> templateTypeRepToText expected <> "\" but it was actually a \"" <> templateTypeRepToText actual <> "\""
-    ContractDoesNotImplementInterface _ t i ->
-      "\"" <> templateTypeRepToText t <> "\" does not implement interface \"" <> templateTypeRepToText i <> "\""
-    ContractDoesNotImplementRequiringInterface _ t required requiring ->
-      "\"" <> templateTypeRepToText t <> "\" does not implement interface \"" <> templateTypeRepToText required <> "\" as required by \"" <> templateTypeRepToText requiring <> "\""
+    CreateEmptyContractKeyMaintainers _ -> "Attempted to create a contract with empty key maintainers"
```
Comment (carlpulley-da) [inline daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:146]: With `AnyTemplate` arguments, there's not a lot that we can do.

File: daml-script/daml3/Daml/Script/Questions/Submit/Error.daml [hunk 2]
```diff
@@ -121,34 +122,38 @@ data SubmitError
...
+instance Show AnyContractKey where
+  show (AnyContractKey _ templateTypeRep) = show templateTypeRep
+
 instance Show SubmitError where
   show err = case err of
     ContractNotFound { unknownContractIds, additionalDebuggingInfo } ->
       "Could not find contract with ID(s) " <> intercalate ", " (toList unknownContractIds) <> "."
       <> foldMap (\info -> " Additional debugging info: " <> show info) additionalDebuggingInfo
-    ContractKeyNotFound {} -> "Could not find contract with given key"
+    ContractKeyNotFound contractKey -> "Could not find contract with given key" <> show contractKey
```
Comment (samuel-williams-da) [inline daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:136]:
```suggestion
    ContractKeyNotFound contractKey -> "Could not find contract with given key: " <> show contractKey
```

File: daml-script/daml3/Daml/Script/Questions/Submit/Error.daml [hunk 3]
```diff
@@ -121,34 +122,38 @@ data SubmitError
     -- TODO: Should we expose this at all?
     TruncatedError with
       truncatedErrorType : Text
-        -- ^ One of the contructor names of SubmitFailure except DevError, UnknownError, TruncatedError
+        -- ^ One of the constructor names of SubmitFailure except DevError, UnknownError, TruncatedError
       truncatedErrorMessage : Text
 
--- TODO[SW]: Add more detail here
+instance Show AnyContractKey where
+  show (AnyContractKey _ templateTypeRep) = show templateTypeRep
```
[the remark this answers is not in the document]
Comment (samuel-williams-da) [reply daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:129]: I think this is reasonable for now, perhaps later we can create a method to stringify AnyContractKey payloads

File: daml-script/daml3/Daml/Script/Questions/Submit/Error.daml (hunk 1 above)
Comment (samuel-williams-da) [reply daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:146]: Thats unfortunate, and ofc in a stable package so we can't do anything without adding a primative/question. I'm okay with this as is.

**Review by samuel-williams-da — APPROVED** [review]

**Comment by samuel-williams-da** [comment]
Thanks Carl!

**Pull request merged**